### DISCUSSION TITLE
Metrics such as the number of queries performed

### DISCUSSION THREAD

**Discussion by svarlet** [simolus3/drift#1760] [Q&A]
Hi,

Someone in my team just asked if drift has a metrics api by which we could find how many queries were performed at arbitrary times. For example, how many queries were executed when a tap on this button rendered that view.

I can't find such a feature in the docs, discussions, or issues, so I imagine the answer is no. In that case, where should I plug a home made "query counter"?

Thanks

**Comment by simolus3** [comment]
You could write a custom [`QueryExecutor`](https://drift.simonbinder.eu/api/backends/queryexecutor-class) for this. It would wrap an existing query executor (like a `NativeDatabase`) and delegate all its work, but keep metrics for each invoked method.

You could take a look at the [`LazyDatabase` implementation](https://github.com/simolus3/drift/blob/6dc198ecca5c003c6c6bcb381dcd63861beba679/drift/lib/src/utils/lazy_database.dart) for inspiration, except that you'll also want to override `beginTransaction()` to return a wrapping `TransactionExecutor` counting queries.
